lara-typescript/src/section-authoring/util/move-utils.spec.ts

```diff
@@ -0,0 +1,123 @@
+import { IPage, ISection, ISectionItem } from "../api/api-types";
+import { findSectionAddress, findSection, findSectionByAddress } from "./finding-utils";
+import { moveSection, ISectionDestination, RelativeLocation } from "./move-utils";
+import { makePages } from "./spec-helper";
+
+describe("moveSection", () => {
+
+  describe("Failing scenarios" , () => {
+    it ("it can't move non-existent sections", () => {
+      const pages = makePages(3);
+      const destination: ISectionDestination = {
+        destPageId: pages[0].id,
+        relativeLocation: RelativeLocation.After,
+        destSectionId: "section3" // not specifying a page...
+      };
+      const sectionId = "bogus";
+
+      // Move failed:
+      expect(moveSection({ destination, pages, sectionId })).toEqual([]);
+    });
+  });
+
+  describe("moving sections in the same page", () => {
+
+    it ("it can move a section to the end of the page with a bogus section name", () => {
+      const pages = makePages(3);
+      const destination: ISectionDestination = {
+        relativeLocation: RelativeLocation.After,
+        destPageId: pages[0].id,
+        destSectionId: "bogus" // not specifying a page...
+      };
+      const sectionId = "section0";
+
+      const changedPages = moveSection({ destination, pages, sectionId });
+      expect(changedPages.length).toEqual(1);
+      expect(changedPages[0].sections.map(s => s.id)).toEqual(["section1", "section2", "section0" ]);
+    });
+
+    it("can move a section ahead within the same page", () => {
+      const pages = makePages(3);
+      const destination: ISectionDestination = {
+        destPageId: pages[0].id,
+        relativeLocation: RelativeLocation.Before,
+        destSectionId: "section0" // not specifying a page...
+      };
+
+      const sectionId = "section1";
+      const changedPages = moveSection({ destination, pages, sectionId });
+      expect(changedPages.length).toEqual(1);
+      expect(changedPages[0].sections.map(s => s.id)).toEqual(["section1", "section0", "section2"]);
+    });
+
+    it("can move a section ahead within the same page", () => {
+      const pages = makePages(3);
+      const destination: ISectionDestination = {
+        destPageId: pages[0].id,
+        relativeLocation: RelativeLocation.Before,
+        destSectionId: "section0" // not specifying a page...
+      };
+
+      const sectionId = "section2";
+      const changedPages = moveSection({ destination, pages, sectionId });
+      expect(changedPages.length).toEqual(1);
+      expect(changedPages[0].sections.map(s => s.id)).toEqual(["section2", "section0", "section1"]);
+
+    });
```

Comment:
Are these tests both moving sections backward instead of forward? If so, should one of them be changed to move a section forward and then the it statements adjusted accordingly? (They say the same right now.)